app/models/course/assessment/assessment_ability.rb

```diff
@@ -104,10 +104,10 @@ def allow_manager_publish_submissions
   def allow_staff_grade_submissions
     can [:read, :update, :reload_answer],
         Course::Assessment::Submission, assessment: assessment_course_staff_hash
-    can :grade, Course::Assessment::Submission, submissions_without_attempting_hash.merge(
+    can :grade, Course::Assessment::Submission, all_submissions_hash.merge(
```

Comment:
This can be merged with the definition above:
` can [:read, :update, :grade, :reload_answer],  Course::Assessment::Submission, assessment: assessment_course_staff_hash`

Think `all_submissions_hash.merge` is not needed any more.